react/features/rtcstats/RTCStats.ts

```diff
@@ -5,174 +5,48 @@ import {
     PC_STATE_FAILED
     // @ts-expect-error
 } from '@jitsi/rtcstats/events';
-// @ts-expect-error
-import rtcstatsInit from '@jitsi/rtcstats/rtcstats';
-// @ts-expect-error
-import traceInit from '@jitsi/rtcstats/trace-ws';
-/* eslint-enable lines-around-comment */
 
-import { createRTCStatsTraceCloseEvent } from '../analytics/AnalyticsEvents';
-import { sendAnalytics } from '../analytics/functions';
+import JitsiMeetJS from '../base/lib-jitsi-meet';
 
 import logger from './logger';
 import {
     DominantSpeakerData,
     E2ERTTData,
     FaceLandmarksData,
-    InitOptions,
     VideoTypeData
 } from './types';
 
 /**
- * Filter out RTCPeerConnection that are created by callstats.io.
- *
- * @param {*} config - Config object sent to the PC c'tor.
- * @returns {boolean}
- */
-function connectionFilter(config: any) {
-    if (config?.iceServers[0] && config.iceServers[0].urls) {
-        for (const iceUrl of config.iceServers[0].urls) {
-            if (iceUrl.indexOf('callstats.io') >= 0) {
-                return true;
-            }
-        }
-    }
-}
-
-/**
- * Class that controls the rtcstats flow, because it overwrites and proxies global function it should only be
- * initialized once.
+ * Handle lib-jitsi-meet rtcstats events and send jitsi-meet specific statistics.
  */
 class RTCStats {
-    trace: any;
-    options?: InitOptions;
-    isPeerConnectionWrapped = false;
-    connStateEvents: any = [];
+    private _connStateEvents: Array<any> = [];
+    private _initialized = false;
 
     /**
-     * Initialize the rtcstats components, if the options have changed.
+     * Handles rtcstats events.
      *
-     * @param {Oject} newOptions -.
      * @returns {void}
      */
-    maybeInit(newOptions: InitOptions) {
-        const oldOptions = this.options;
-        const changed = !oldOptions || (oldOptions.endpoint !== newOptions.endpoint
-            || oldOptions.meetingFqn !== newOptions.meetingFqn
-            || oldOptions.pollInterval !== newOptions.pollInterval
-            || oldOptions.sendSdp !== newOptions.sendSdp
-            || oldOptions.useLegacy !== newOptions.useLegacy);
+    init() {
+        this._connStateEvents = [];
 
-        if (changed) {
-            this.reset();
-
-            if (newOptions.meetingFqn && newOptions.endpoint) {
-                this.init(newOptions);
-            } else {
-                logger.warn('RTCStats is enabled but it has not been configured.');
-            }
+        if (!this._initialized) {
+            JitsiMeetJS.rtcstats.events.on(
```

Comment:
As discussed on the other PR, let's use constants for event names and the `rtcstats` itself be an event emitter.

Reply:
Adjusted accordingly.